### ISSUE TITLE
dont do this yet

### ISSUE THREAD

**Issue by ed588** [werewolves-devs/Werewolf_Bot#3]
@Randium check the discord, we are deciding how to do this - don't start doing this yet

**Comment by HurricanKai** [comment]
I don't quite think he cares that much lol, I think he just wants this to be done quick, and doesn't give a shit about our development concepts

**Comment by ed588** [comment]
> Understandable, but not good...
Exactly. If he rushes this it won't be as good and it won't be maintainable. Much better to wait for everyone to agree how this should be done and then work on it.
@Randium

**Comment by Randium** [comment]
Somehow I didn't receive a notification from that mention. But sure, I'll look into the Discord.

Keep in mind that the stuff I'm currently writing isn't permanent. I'm just farting out some ideas, and I wanted to see what you thought about it.

**Comment by ed588** [comment]
> Keep in mind that the stuff I'm currently writing isn't permanent. I'm just farting out some ideas, and I wanted to see what you thought about it.

what you've done so far seems good - one of the things i'd like to improve from the last one is better use of classes and OO. And I think python might be a better choice for the main bot after all

**Comment by ed588** [comment]
saying that, I don't think roles should be hardcoded. What i do think we should do is have roles be more class based, which would save time and also automatically do some stuff that we were doing manually before.

**Comment by HurricanKai** [comment]
I woud suggest an "Register Role" Function, keeping Track of any Roles Reistered, which the bot then can Call Functions on. i dont know how far that is possible in Python, but if possible, have a base class "Role" which any Registered Role has to inherit, and then "virtual" Functions, meaning Functions that either do there base thing (defined in Role) or the overriden thing in the Specific Role Class
